src/api-extension/cart.service.ts

```diff
@@ -203,28 +203,71 @@ export class CartService {
 
     coupons = this.filterCouponsByLimit(coupons, couponsLimit);
 
-    const validatedCoupons =
+    let validatedCoupons =
       await this.voucherifyConnectorService.validateStackableVouchersWithCTCart(
         coupons.filter((coupon) => coupon.status != 'DELETED'),
         cart,
         this.productMapper.mapLineItems(cart.lineItems),
         sessionKey,
       );
 
+    const productsToAdd = await convertUnitTypeCouponsToFreeProducts(
+      validatedCoupons,
+      this.commerceToolsConnectorService.getClient(),
+      this.getPriceSelectorFromCart(cart),
+    );
+
+    const productsToChange = productsToAdd.filter(
+      (product) => product.discount_difference !== 0,
+    );
+
+    if (productsToChange) {
+      const productsToChangeSKUs = productsToChange.map(
+        (productsToChange) => productsToChange.product,
+      );
+      let items = validatedCoupons.order.items;
+      items = items.map((item: any) => {
```

Comment:
According to ValidationValidateStackableResponse type `item` type should be `OrdersItem` so `item.discount_quantity`, `item.sku.price` and `(item.sku as any).source_id)` are no present here. Please double check if I'm right.

Reply:
item.discount_quantity, item.sku.price and (item.sku as any).source_id) does exist for some reason. This may be V% typing error.